src/gui/controllers/TaskController.java

```diff
@@ -18,5 +22,25 @@
 	public void initialize(URL location, ResourceBundle resources) {
 
 	}
+	
+	@FXML public void accept(ActionEvent event) {
+		Object userData = textFieldTitle.getScene().getWindow().getUserData();
+		if (isValidData() && userData != null && userData instanceof TasksHandler) {
+			try {
+				((TasksHandler) userData).addTask(new Task(textFieldTitle.getText(),textAreaBody.getText()));
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+		Utils.closeWindow(event);
```

Comment:
I think this line should be executed only if the task is saved.
Move this line between line 30 and 31.

Reply:
Done